l10n_it_esigibilita_iva/__manifest__.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2017 Alessandro Camilli - Openforce
+# Copyright 2018 Lorenzo Battistini <https://github.com/eLBati>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+{
+    "name": "Italian Localization - Esigibilita' IVA",
```

Comment:
Puoi cogliere l'occasione per adattarlo a https://www.odoo-italia.org/documentazione/12.0/sviluppo/guidelines.html#titolo-del-modulo ?

Reply:
Il passaggio alla 14.0 potrebbe essere l'occasione giusta per uniformare i nomi dei moduli e portare in inglese i pochi rimasti in italiano. In questo caso *l10n_it_vat_payability*.
Questo anche in ottica di collaborazione più ampia a livello internazionale (anche con Odoo S.A. ad esempio).

Che ne pensate @OCA/local-italy-developers @OCA/local-italy-maintainers ?

Reply:
@primes2h +1

Reply:
Sono a favore.  Mi chiedo però se abbia senso anche prevedere uno script di migrazione o una procedura simile a quella descritta in https://github.com/OCA/l10n-italy/blob/12.0/l10n_it_invoices_data_communication/README.rst#known-issues--roadmap in modo da ripulire i dati che fanno riferimento al vecchio nome.

Reply:
@tafaRU Completamente d'accordo sul dettagliare la parte della migrazione

Reply:
La procedura `update_module_names` è necessaria, se si cambia nome.

Questo vale ovviamente per tutti i moduli che verranno eventualmente rinominati

Reply:
Il punto è che il caso descritto in https://github.com/OCA/l10n-italy/blob/12.0/l10n_it_invoices_data_communication/README.rst#known-issues--roadmap è diverso da quello che abbiamo qui. Ci vorrebbe pertanto qualche volontario che verifichi qual è la soluzione da adottare.

Reply:
@eLBati 

> La procedura update_module_names è necessaria, se si cambia nome.

e anche sufficiente? (vedi https://github.com/OCA/l10n-italy/pull/1938#discussion_r531689275)

Reply:
Non lo so, bisogna provare

Reply:
Anche secondo me serve uno script di migrazione che esegua https://github.com/OCA/openupgradelib/blob/01f33c913ff08f8b822129a7bc2b7a39200dabc6/openupgradelib/openupgrade.py#L1286 quando si cambia nome del modulo.

> e anche sufficiente? (vedi https://github.com/OCA/l10n-italy/pull/1938#discussion_r531689275)

non ho capito, vuoi sapere se quando eseguo `update_module_names` allora è necessario anche cambiare il nome del modulo?

> Non lo so, bisogna provare

Posso provare, ma non avendo capito quanto sopra, non so quale sarebbe la prova da fare; mi potete dare degli step?

Riguardo https://github.com/OCA/l10n-italy/blob/12.0/l10n_it_invoices_data_communication/README.rst#known-issues--roadmap, secondo me serviva uno script di migrazione del modulo di modo che quello script venisse eseguito automaticamente

Reply:
Se non ricordo male, il motivo per cui, nel caso di `l10n_it_comunicazione_dati_iva`, la procedura è stata messa nel README è che `l10n_it_comunicazione_dati_iva` non era su un repository OCA, ma esterno.

Nel caso di `l10n_it_esigibilita_iva` sì, serve uno script  che venga eseguito automaticamente

Reply:
Ho fatto un primo tentativo, lo trovate qui: https://github.com/Borruso/l10n-italy/pull/4

Non so se va bene ed è sufficiente allo scopo. Se qualcuno riesce a provarlo poi lo potremmo estendere anche agli altri moduli.

Reply:
@Borruso https://github.com/Borruso/l10n-italy/pull/4 è stata testata prima di mergiarla?
Vedi https://github.com/Borruso/l10n-italy/pull/4/files#r561897583

Reply:
Bisogna capire come poter testare lo script di migrazione. A tal proposito incollo il link all'inizio della discussione affrontata sul canale Discord: https://discord.com/channels/753902328494424064/753902328494424070/802132198555516959

Reply:
Qualcuno può chiedere consiglio su contributors@odoo-community.org ?

Reply:
https://odoo-community.org/groups/contributors-15/contributors-185892

Reply:
Qui c'è un esempio pratico di una variazione del nome di un modulo nella migrazione alla v.13.0.

È partito tutto da qui https://github.com/OCA/manufacture/pull/481#issuecomment-730439422

Questa la PR https://github.com/OCA/manufacture/pull/595 nella quale non hanno inserito script di migrazione (*) ma questo commento di Pedro mi sembra importante https://github.com/OCA/manufacture/pull/595#issuecomment-778635409

(*) forse la gestiscono direttamente in OpenUpgrade?

Reply:
Per completezza, visto che su https://odoo-community.org/groups/contributors-15/contributors-185922 non si legge, riporto la risposta di Holger:

> > In general, is it possible to test v14 upgrade scripts before OpenUpgrade is
> >
> > available for v14?
>
> yes, you don't need OpenUpgrade for that.
>
> We lack a good mechanism to test migration scripts, I'm also not sure how I 
would implement such a thing in general. The specific script looks a bit 
pointless to test to me, but you could mock the openupgradelib calls and call 
the function if you're into coverage